packages/graphql/src/schema/validation/validate-document.ts

```diff
@@ -216,19 +221,69 @@ function filterDocument(document: DocumentNode, features: Neo4jFeaturesSettings
     };
 }
 
-function getBaseSchema({
+function runValidationRulesOnFilteredDocument({
+    schema,
+    document,
+    extra,
+    callbacks,
+}: {
+    schema: GraphQLSchema;
+    document: DocumentNode;
+    extra?: {
+        enums: EnumTypeDefinitionNode[];
+        interfaces: InterfaceTypeDefinitionNode[];
+        unions: UnionTypeDefinitionNode[];
+        objects: ObjectTypeDefinitionNode[];
+    };
+    callbacks?: Neo4jGraphQLCallbacks;
+}) {
+    // ==================== for rules testing ====
+    const errors = validateSDL(
+        document,
+        [
+            ...specifiedSDLRules,
+            DirectiveIsValid(extra, callbacks),
+            DirectiveCombinationValid(),
+            SchemaOrTypeDirectives(),
+            ValidJwtDirectives(),
+            ValidGlobalID(),
+            ValidRelationshipProperties(),
+            ValidFieldTypes(),
+            ReservedTypeNames(),
+            ValidObjectType(),
+            ValidDirectiveInheritance(),
+            DirectiveArgumentOfCorrectType,
+        ],
+        schema
+    );
+    const filteredErrors = errors.filter((e) => e.message !== "Query root type must be provided.");
+    if (filteredErrors.length) {
+        throw filteredErrors;
+    }
+    // ===========================================
+}
+
+function validateDocument({
     document,
     features,
     additionalDirectives = [],
     additionalTypes = [],
+    extra,
+    callbacks,
 }: {
     document: DocumentNode;
     features: Neo4jFeaturesSettings | undefined;
-    additionalDirectives: Array<GraphQLDirective>;
-    additionalTypes: Array<GraphQLNamedType>;
-}): GraphQLSchema {
-    const doc = filterDocument(document, features);
-
+    additionalDirectives?: Array<GraphQLDirective>;
+    additionalTypes?: Array<GraphQLNamedType>;
+    extra?: {
+        enums: EnumTypeDefinitionNode[];
+        interfaces: InterfaceTypeDefinitionNode[];
+        unions: UnionTypeDefinitionNode[];
+        objects: ObjectTypeDefinitionNode[];
+    };
```

Comment:
This might change with the schema generation work but do you have any suggestions in the mean time?

Reply:
I would perhaps bundle them all under a single object? Because the purpose is essentially the same right?

```
additionalDefinitions?: {
    additionalDirectives?: Array<GraphQLDirective>;
    additionalTypes?: Array<GraphQLNamedType>;
    enums?: EnumTypeDefinitionNode[];
    interfaces?: InterfaceTypeDefinitionNode[];
    unions?: UnionTypeDefinitionNode[];
    objects?: ObjectTypeDefinitionNode[];
}
```